Configuration/PyReleaseValidation/python/relval_steps.py

```diff
@@ -2789,6 +2789,12 @@ def gen2018HiMix(fragment,howMuch):
     '--filein':'file:step3_inMINIAODSIM.root',
     '--era':'Run2_2016',
     }
+steps['NANOUP15Had']=merge([{'--filein':'file:step4_inMINIAODSIM.root'},steps['NANOUP15']])
+steps['NANOUP17']=merge([{'--conditions':'auto:phase1_2017_realistic','--era' : 'Run2_2017','--geometry' : 'DB:Extended'},steps['NANOUP15']])
+steps['NANOUP17Had']=merge([{'--filein':'file:step4_inMINIAODSIM.root'},steps['NANOUP17']])
+steps['NANOUP18']=merge([{'--conditions':'auto:phase1_2018_realistic','--era' : 'Run2_2018','--geometry' : 'DB:Extended'},steps['NANOUP15']])
+steps['NANOUP18Had']=merge([{'--filein':'file:step4_inMINIAODSIM.root'},steps['NANOUP18']])
+steps['NANOUP18ml']=merge([concurrentLumis,steps['NANOUP18']])
```

Comment:
@ahmad3213 
I just realised that what you have added (also NANOUP15) somehow duplicated with what we already have down there around L2981. Can you make use of those steps?

Reply:
@ahmad3213 
Indeed. And one of topic we discussed in brainstorming tools development is to try cleaning up this code in the long term, so we shouldn't increase the dirtiness. I just realised what you defined here in #28061 "NANOUP15" is similar as what we have already had https://github.com/ahmad3213/cmssw/blob/7fb792de6862c3a261dc341ab595af0f8affd1c6/Configuration/PyReleaseValidation/python/relval_steps.py#L2978 
Can you redefine NANOUP15 to something like

steps['NANOUP15'] = merge([{ '--conditions':'auto:run2_mc', '--era':'Run2_2016', '--filein':'file:step3_inMINIAODSIM.root'}, stepNanoEDMMCProd ])
steps['NANOUP17'] = merge([{'--conditions':'auto:phase1_2017_realistic','--era' : 'Run2_2017', filein':'file:step3_inMINIAODSIM.root', '--geometry' : 'DB:Extended'}, stepNanoEDMMCProd])
and similarly to steps['NANOUP15Had'], steps['NANOUP17Had']

For NANOUP18, I think it's exactly same as steps['NANOEDMMC2018_PROD'] which is there already, no? Since it's not used, we can rename it to NANOUP18. 

PS. Please add these steps in the region of NanoAOD in the code as well (under this line probably https://github.com/ahmad3213/cmssw/blob/7fb792de6862c3a261dc341ab595af0f8affd1c6/Configuration/PyReleaseValidation/python/relval_steps.py#L3005)

Reply:
> we shouldn't increase the dirtiness.

I second this comment from @chayanit - any duplication of code increases the maintenance burden and possible sources of confusion.

Reply:
i agree and followed (did not noticed that lines) in first look during last #28061 .